paddle/framework/backward.cc

```diff
@@ -368,7 +368,7 @@ std::vector<std::unique_ptr<OpDescBind>> MakeBlockBackward(
     ProgramDescBind& program_desc, int block_idx,
     std::unordered_set<std::string>* no_grad_vars,
     std::unordered_map<std::string, std::string>* grad_to_var) {
-  BlockDescBind* cur_block = program_desc.Block(block_idx);
+  BlockDescBind* cur_block = program_desc.MutableBlock(block_idx);
```

Comment:
It looks to me that we don't have to give prefix "mutable" to C++ methods because we can have re-use the same method name with different signature?

```c++
class Some {
 public:
  Block* Block();
  const Block& Block() const;
};
```

I know that we have `Tensor::mutable_data` and `Tensor::data`, but they have different syntax -- the former is used to (re-)allocate memory, whereas the latter doesn't.

In this case, it seems that we can just name it Block, other than MutableBlock?

Reply:
```cpp
#include <iostream>

class Cls {
public:
  int a_;
  Cls(int a) : a_(a) {}

  int *Data() { return &a_; }

  const int &Data() const { return a_; }
};

int main() {
  Cls instance(10);
  std::cout << instance.Data() << std::endl;
  return 0;
}
```

Reply:
If we do not use `MutableData`, it may be a little bit confusing for users.
As the code before, the std::cout will print the pointer of a, not the value of a.